src/main/java/de/srendi/advancedperipherals/common/util/CoordUtil.java

```diff
@@ -48,31 +48,37 @@ public static boolean isInRange(@NotNull BlockPos blockPos, @NotNull Player play
         return world.getNearbyPlayers(TargetingConditions.forNonCombat(), null, new AABB(firstPos, secondPos)).contains(player);
     }
 
-    public static Direction getDirection(FrontAndTop orientation, String computerSide) throws LuaException {
-        if (Direction.byName(computerSide) != null) return Direction.byName(computerSide);
+    public static Direction getDirection(FrontAndTop orientation, final String computerSide) throws LuaException {
+        if (computerSide == null) {
+            throw new LuaException("null is not a valid side");
+        }
+
+        final String side = computerSide.toLowerCase();
```

Comment:
Is there a particular reason you did this?
re-asigning the var would be enough and would remove one unnecessary var

Reply:
> re-asigning the var would be enough and would remove one unnecessary var

I would like to throw the origin side that passed in when the side is not exists, but seems you already make them upperCase Idk why

Reply:
I removed line 135, if you think the exception should be case insensitive or always be uppercase, you can commit to the branch I think, or you can reopen this conversation

Reply:
Uhm, I didn't even see that

I mean the initialization of the var `side`

So instead of
```java
    public static Direction getDirection(FrontAndTop orientation, final String computerSide) throws LuaException {
        if (computerSide == null) {
            throw new LuaException("null is not a valid side");
        }

        final String side = computerSide.toLowerCase(Locale.ROOT);
```

You do
```java
    public static Direction getDirection(FrontAndTop orientation, String computerSide) throws LuaException {
        if (computerSide == null) {
            throw new LuaException("null is not a valid side");
        }

       computerSide = computerSide.toLowerCase(LocaleRoot);
```

And remove `side` completely which would remove one unnecessary var